### PR TITLE
account for case where only doi_preprint is set

### PR DESCRIPTION
**Pull request by le-ander**
Many thanks for contributing to sfaira!

**PR Checklist**
Please fill in the appropriate checklist below (delete whatever is not relevant). These are the most common things requested on pull requests (PRs).

 - [x] This comment contains a description of changes (with reason)
 - [x] Referenced issue is linked

**Description of changes**
fixes the error described in #695 when the sfaira universe is subset for by the `doi` key and only doi_prepring is set for a dataset.

### CODE REVIEW

**Comment by le-ander** [comment]
closes #695

**Comment by le-ander** [comment]
do you think this is the right way to deal with the error @davidsebfischer ?

File: sfaira/data/dataloaders/base/dataset_group.py [hunk 1]
```diff
@@ -486,7 +486,7 @@ def doi(self) -> List[str]:
         """
         dois = []
         for _, v in self.datasets.items():
-            vdoi = v.doi_journal
+            vdoi = v.doi_journal if v.doi_journal is not None else v.doi_preprint
```
Comment (davidsebfischer) [inline sfaira/data/dataloaders/base/dataset_group.py:489]: Is this not equivalent to `vdoi = v.doi`? That would be preferable as it pushes DOI management into the Dataset class and thus makes it more centralized.

Comment (le-ander) [reply sfaira/data/dataloaders/base/dataset_group.py:489]: of course, that make so much more sense!

**Comment by davidsebfischer** [comment]
Thanks!

**Pull request merged**